lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -1365,11 +1376,11 @@ txConstraints protocolParams witnessTag = TxConstraints
     txOutputBelowMinimumAdaQuantity =
         isBelowMinimumCoinForUTxO (minimumUTxO protocolParams)
 
-    txRewardWithdrawalCost c =
-        marginalCostOf empty {txRewardWithdrawal = c}
+    txRewardWithdrawalCost =
+        feeOf . txRewardWithdrawalSize
 
-    txRewardWithdrawalSize c =
-        marginalSizeOf empty {txRewardWithdrawal = c}
+    txRewardWithdrawalSize (Coin 0) = TxSize 0
+    txRewardWithdrawalSize _ = TxSize sizeOfSignedWithdrawal
```

Comment:
I'm missing the constant overhead here, which is either good or bad, and most likely bad 🤔